packages/core/src/plugins/QueueFlushingPlugin.ts

```diff
@@ -60,6 +73,19 @@ export class QueueFlushingPlugin extends UtilityPlugin {
    * Calls the onFlush callback with the events in the queue
    */
   async flush() {
+    // Wait for the queue to be restored
+    try {
+      await this.isRestored;
+    } catch (e) {
+      // If the queue is not restored before the timeout, we will notify but not block flushing events
```

Comment:
Any chance we get another type of error here that is not related to timeouts? Should those be handled any differently?

Reply:
Not really, any other issue during restoration will be handled elsewhere. JS is notoriously limited in error handling.